### PR TITLE
[WP Individual JP Plugin] Show dialog with overlay

### PR DESCRIPTION
**Pull request by thomashorta**
Part of #18114

Create a dialog for showing the Overlay screen. (only in the WordPress app)

The dialog is shown on top of the following screens:
- Site Picker
- Login Epilogue (site picker after login)
- Post Signup Interstitial (shown after first login ever - clear data - with no WP.com sites in the account)
- My site screen with empty state (after login with no WP.com sites in the account)

When the WP.com account has at least 1 "problem site", which fits the following conditions:
- self-hosted site 
- has at least 1 Individual Jetpack plugin active and connected to the WP.com account 
- does NOT have the full Jetpack plugin installed

## To test:
Here are some of the test scenarios (WordPress app only).

### Login - At least 1 WP.com site + at least 1 "problem site"
1. Log in to WordPress
2. **verify** the Login Epilogue (site picker) shows only WP.com sites
3. **verify** the overlay saying there are other sites with individual JP plugins is shown

### Site Picker - At least 1 WP.com site + at least 1 "problem site"
1. After selecting one WP.com site
2. tap the Site selection arrow to open site picker
3. **verify** the site picker shows only WP.com sites
4. **verify** the overlay saying there are other sites with individual JP plugins is shown

### Login - NO WP.com sites + at least 1 "problem site"
Important: this scenario only happens once, so to see it the app should be in a clear state by clearing the app data in the device settings. 

1. Log in to WordPress
2. **verify** the **Post Signup Interstitial** is shown saying you have NO sites
3. **verify** the overlay saying there are sites with individual JP plugins is shown

### My Site - NO WP.com sites + at least 1 "problem site"
1. After login, skip the **Post Signup Interstitial**
2. select the **My Site tab** (bottom navigation bar)
3. **verify** the overlay saying there are sites with individual JP plugins is shown

## Notes
There are different overlay contents for each situation below. It would be good to test all combinations, but it can be complicated getting accounts in those specific situations:
1. Single "problem site" with a single JP Individual plugin
2. Single "problem site" with multiple JP individual plugins
3. Multiple "problem sites" some with single JP individual plugins and some with multiple plugins 

Currently there is no logic to avoid showing the overlay repeatedly, which will be added in https://github.com/wordpress-mobile/WordPress-Android/issues/18175.

## Regression Notes
1. Potential unintended areas of impact
Full-screen bottoms sheets behavior (ex: theme preview, blocs preview, jetpack feature overlay).

10. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

11. What automated tests I added (or what prevented me from doing so)
Added tests to all viewmodels showing the new overlay.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginEpilogueViewModel.kt [hunk 1]
```diff
@@ -52,4 +56,21 @@ class LoginEpilogueViewModel @Inject constructor(
...
     }
+
+    fun onSiteListLoaded() {
+        // don't check if already shown
+        if (_navigationEvents.value?.peekContent() == LoginNavigationEvents.ShowJetpackIndividualPluginOverlay) return
+
+        viewModelScope.launch {
+            val showOverlay = wpJetpackIndividualPluginHelper.shouldShowJetpackIndividualPluginOverlay()
+            if (showOverlay) {
+                delay(DELAY_BEFORE_SHOWING_JETPACK_INDIVIDUAL_PLUGIN_OVERLAY)
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginEpilogueViewModel.kt:67]: ❓ I was wondering: why do we need this delay before posting the overlay event? 🤔

File: WordPress/src/main/java/org/wordpress/android/util/extensions/DialogExtensions.kt [hunk 2]
```diff
@@ -26,3 +31,22 @@ fun Dialog.setStatusBarAsSurfaceColor() {
         }
     }
 }
+
+fun BottomSheetDialog.fillScreen(isDraggable: Boolean = false) {
+    setOnShowListener {
+        val bottomSheet: FrameLayout = findViewById(
+            com.google.android.material.R.id.design_bottom_sheet
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/util/extensions/DialogExtensions.kt:38]: ❓ I was wondering if we should try to get this View without relying on the ID from the library 🤔 do you think that's possible/makes sense?

File: WordPress/src/main/java/org/wordpress/android/viewmodel/accounts/PostSignupInterstitialViewModel.kt [hunk 3]
```diff
@@ -52,5 +59,27 @@ class PostSignupInterstitialViewModel
...
 
-    enum class NavigationAction { START_SITE_CREATION_FLOW, START_SITE_CONNECTION_FLOW, DISMISS }
+    private fun checkJetpackIndividualPluginOverlayShouldShow() {
+        // don't check if already shown
+        if (navigationAction.value == SHOW_JETPACK_INDIVIDUAL_PLUGIN_OVERLAY) return
+
+        viewModelScope.launch {
+            val showOverlay = wpJetpackIndividualPluginHelper.shouldShowJetpackIndividualPluginOverlay()
+            if (showOverlay) {
+                delay(DELAY_BEFORE_SHOWING_JETPACK_INDIVIDUAL_PLUGIN_OVERLAY)
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/viewmodel/accounts/PostSignupInterstitialViewModel.kt:69]: ❓ Same question about the delay before posting the overlay event

File: WordPress/src/main/java/org/wordpress/android/viewmodel/main/SitePickerViewModel.kt [hunk 4]
```diff
@@ -66,10 +58,24 @@ class SitePickerViewModel @Inject constructor(
...
     }
 
+    fun onSiteListLoaded() {
+        // don't check if already shown
+        if (_onActionTriggered.value?.peekContent() == ShowJetpackIndividualPluginOverlay) return
+
+        viewModelScope.launch {
+            val showOverlay = wpJetpackIndividualPluginHelper.shouldShowJetpackIndividualPluginOverlay()
+            if (showOverlay) {
+                delay(DELAY_BEFORE_SHOWING_JETPACK_INDIVIDUAL_PLUGIN_OVERLAY)
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/viewmodel/main/SitePickerViewModel.kt:68]: ❓ Same question about the delay before posting the overlay event

**Review by RenanLukas — COMMENTED** [review]
Thanks for the changes, @thomashorta . All the scenarios are working as expected 🎉 

I've only left a few questions.

File: WordPress/src/main/java/org/wordpress/android/util/extensions/DialogExtensions.kt (hunk 2 above)
Comment (thomashorta) [reply WordPress/src/main/java/org/wordpress/android/util/extensions/DialogExtensions.kt:38]: Unfortunately, I tried looking into other ways of doing it but this seems to be the most recommended and precise way of getting the correct Layout. It is 1. how I found it over the internet on SO and other forums/posts, and 2. how we are doing it all over the codebase in general, so I just extracted the code to be just called when needed.

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginEpilogueViewModel.kt (hunk 1 above)
Comment (thomashorta) [reply WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginEpilogueViewModel.kt:67]: Without the delay, on my tests, the user experience suffers as you don't even see the screen you arrived at and this other screen pops up to view almost at the same time. 

I only added a delay to improve that experience, so you can properly see the underlying screen appear, and half a second later the modal. We can check with @osullivanchris if this makes sense, I will try to record a video later today.

**Review by RenanLukas — APPROVED** [review]
LGTM :shipit:

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginEpilogueViewModel.kt (hunk 1 above)
Comment (thomashorta) [reply WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginEpilogueViewModel.kt:67]: Here are a couple of videos, with and without the delay, for comparison.

<details><summary>With delay</summary>

https://user-images.githubusercontent.com/5091503/228273531-2168c950-200a-413b-9a07-8a85703faa43.mp4

</details> 

<details><summary>Without delay</summary>

https://user-images.githubusercontent.com/5091503/228275557-cbc125e6-702b-4aa5-8d32-85f05b77344f.mp4

</details> 

I will merge this PR for now, but @osullivanchris please feel free to take a look and provide feedback if the delay looks bad or something and I can adjust it in future PRs. 

My personal opinion is that the delay just makes things feel more intentional: first, briefly show the screen the user intended to go then show the warning modal with extra info for this context.

Comment (osullivanchris) [reply WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginEpilogueViewModel.kt:67]: @thomashorta I think the delay is good. I agree, and would take same approach 👍

**Pull request merged**